test/e2e/scrapeconfig_test.go

```diff
@@ -0,0 +1,263 @@
+// limitations under the License.
+
+package e2e
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	monitoringv1alpha1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1alpha1"
+)
+
+// testScrapeConfigCreation tests multiple ScrapeConfig definitions
+func testScrapeConfigCreation(t *testing.T) {
+	skipPrometheusTests(t)
+	t.Parallel()
+
+	fiveMins := monitoringv1.Duration("5m")
+
+	tests := []struct {
+		name          string
+		spec          monitoringv1alpha1.ScrapeConfigSpec
+		expectedError bool
+	}{
+		{
+			name: "empty-scrape-config",
+			spec: monitoringv1alpha1.ScrapeConfigSpec{},
+		},
+		{
+			name: "static-config",
+			spec: monitoringv1alpha1.ScrapeConfigSpec{
+				StaticConfigs: []monitoringv1alpha1.StaticConfig{
+					{
+						Targets: []monitoringv1alpha1.Target{"target1:9090", "target2:9090"},
+						Labels: map[monitoringv1.LabelName]string{
+							"label1": "value1",
+							"label2": "value2",
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "http-sd-config",
+			spec: monitoringv1alpha1.ScrapeConfigSpec{
+				HTTPSDConfigs: []monitoringv1alpha1.HTTPSDConfig{
+					{
+						URL:             "http://localhost:8080/file.json",
+						RefreshInterval: &fiveMins,
+					},
+				},
+			},
+		},
+		{
+			name: "file-sd-config",
+			spec: monitoringv1alpha1.ScrapeConfigSpec{
+				FileSDConfigs: []monitoringv1alpha1.FileSDConfig{
+					{
+						Files: []monitoringv1alpha1.SDFile{
+							"/etc/prometheus/sd/file.json",
+							"/etc/prometheus/sd/file.yaml",
+						},
+						RefreshInterval: &fiveMins,
+					},
+				},
+			},
+		},
+		{
+			name: "invalid-sd-config",
+			spec: monitoringv1alpha1.ScrapeConfigSpec{
+				FileSDConfigs: []monitoringv1alpha1.FileSDConfig{
+					{
+						Files: []monitoringv1alpha1.SDFile{
+							"/etc/prometheus/sd/file.invalid",
+						},
+						RefreshInterval: &fiveMins,
+					},
+				},
+			},
+			expectedError: true,
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+			testCtx := framework.NewTestCtx(t)
+			defer testCtx.Cleanup(t)
+			ns := framework.CreateNamespace(context.Background(), t, testCtx)
+
+			sc := framework.MakeBasicScrapeConfig(ns, "scrape-config")
+			sc.Spec = test.spec
+			_, err := framework.MonClientV1alpha1.ScrapeConfigs(ns).Create(context.Background(), sc, metav1.CreateOptions{})
+
+			if err == nil {
+				require.Falsef(t, test.expectedError, "expected error but got nil")
+				return
+			}
+
+			require.Truef(t, apierrors.IsInvalid(err), "expected Invalid error but got %v", err)
+
+			require.True(t, test.expectedError)
```

Comment:
WDYT if we change it like this? What prompted my comment was this principle https://gist.github.com/adamveld12/c0d9f0d5f0e1fba1e551#indent-error-flow
```suggestion
			if test.expectedError {
			        require.Error(t, err)		
  				require.Truef(t, apierrors.IsInvalid(err), "expected Invalid error but got %v", err)
				return
			}
			require.NoError(t, err)
			require.Falsef(t, test.expectedError, "expected error but got nil")
```

Reply:
lgtm!